### PR TITLE
add user resource

### PR DESCRIPTION
**Pull request by nonylene**
# example

```ruby
user "example" do
  username "example" # optional
  gid "example_group" # optional
  home "/home/example" # optional
  password "xxxxx" # optional
  system_user false # optional
  uid 1111 # optional
  shell "/bin/bash" # optional
  create_home false # optional
end
```

cf. https://github.com/itamae-kitchen/itamae/blob/master/lib/itamae/resource/user.rb

### CODE REVIEW

**Comment by k0kubun** [comment]
> 6cd3125

Please rebase instaed of merge

**Comment by nonylene** [comment]
I 've rebased and force-pushed.

File: mrblib/itamae/resource_executor/user.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
+module Itamae
+  module ResourceExecutor
+    class User < Base
+      def apply(current, desired)
+        if desired.exist
+          if exist?(desired.username)
```
Comment (k0kubun) [inline mrblib/itamae/resource_executor/user.rb]: Thank you for keeping codebase. While this code is acceptable, it'd be better to use `current.exist` here.

Comment (nonylene) [inline mrblib/itamae/resource_executor/user.rb]: fixed in https://github.com/k0kubun/itamae-mruby/pull/3/commits/dccd0cb4b41dd92c3642806629909366712aaffa

**Comment by k0kubun** [comment]
LGTM. Thank you!

**Pull request merged**